tasks/theme.js

```diff
@@ -47,14 +51,15 @@ module.exports = function(grunt) {
           }, options)
         });
 
-        steps.push('compass:' + key);
+        task = theme.proxy ? 'theme:' + key + ':compass' : 'compass:' + key;
```

Comment:
Check for proxy status above the compass attribute check. If so, add `grunt theme:key:compile` to the steps array. Skip the rest of the loop.